src/components/Tabs/Stability/index.tsx

```diff
@@ -0,0 +1,346 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-param-reassign */
+import { FC, useState, useEffect, MouseEvent } from 'react';
+import { upperCase } from 'lodash';
+import { showErrorNotification } from '@corva/ui/utils';
+
+import { styles } from './styles';
+
+import { CardGroup } from '@/components/common/CardGroup';
+import { Card } from '@/components/common/Card';
+import { ArrowIcon } from '@/components/common/ArrowIcon';
+import { ArrowLineLeftSeafloor } from '@/components/common/icons/ArrowLineLeftSeafloor';
+import { ArrowFourStability } from '@/components/common/icons/ArrowFourStability';
+import { ArrowLineRightStability } from '@/components/common/icons/ArrowLineRightStability';
+import { Popover } from '@/components/common/Popover';
+import {
+  CHIPS,
+  COMMENTS,
+  MUSTER_ABANDON,
+  STABILITY_TAB,
+  STATUS,
+  SUB_DATA,
+  SURFACE_TAB,
+  lossOfModeTabData,
+  surfaceTabData,
+} from '@/constants/data.const';
+import { TChip, TSource, TTabs } from '@/types/global.type';
+import { getComments } from '@/entities/record';
+import { useSubscription } from '@/shared/hooks/useSubscription';
+
+export const StabilityTab: FC<TTabs> = ({ currentUser, rigId, well }) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const [title, setTitle] = useState<string | null>(null);
+  const [source, setSource] = useState<TSource>(lossOfModeTabData);
+  const [dataId, setDataId] = useState<string>();
+  const [surfaceSource, setSurfaceSource] = useState<TSource>(surfaceTabData);
+  const [surfaceDataId, setSurfaceDataId] = useState<string>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const [stabilityData, surfaceData] = await Promise.all([
+          getComments(currentUser.company_id, rigId, STABILITY_TAB),
+          getComments(currentUser.company_id, rigId, SURFACE_TAB),
+        ]);
+        if (stabilityData && stabilityData?.[0]?.data[STABILITY_TAB]) {
```

Comment:
```suggestion
        if (stabilityData && stabilityData?.[0]?.data.?[STABILITY_TAB]) {
```
Similar ones are existed at several parts. they are many so I am difficult to write at all places. please check.